cirrus/google_cloud/manager.py

```diff
@@ -44,6 +44,18 @@
     'STANDARD'  # alias for MULTI_REGIONAL/REGIONAL, based on location
 ]
 
+COMPUTE_ENGINE_DEFAULT = 'COMPUTE_ENGINE_DEFAULT'
+GOOGLE_API = 'GOOGLE_API'
+COMPUTE_ENGINE_API = 'COMPUTE_ENGINE_API'
+USER_MANAGED = 'USER_MANAGED'
+
+GOOGLE_SERVICE_ACCOUNT_DOMAIN_TYPE_MAPPING = [
+    ('appspot.gserviceaccount.com', COMPUTE_ENGINE_DEFAULT),
+    ('cloudservices.gserviceaccount.com', GOOGLE_API),
+    ('compute-system.iam.gserviceaccount.com', COMPUTE_ENGINE_API),
+    ('', USER_MANAGED),
```

Comment:
I don't think this will correctly get user managed service accounts. If I create a service account I get something like: `test@google_project_id.iam.gserviceaccount.com`. did you find any google docs about identifying User Managed emails?